components/canvas/ContextMenu.tsx

```diff
@@ -0,0 +1,116 @@
+import { NodeData } from "@/types/NodeData";
+import { getNodeTypeName } from "@/utils/getNodeTypeName";
+import { capitalizeFirstLetter } from "@/utils/stringHelpers";
+import { Menu } from "@equinor/eds-core-react";
+import { RefObject, useState } from "react";
+import { Node, Position } from "reactflow";
+import { MenuItemExandable } from "../MenuItemExandable";
+import styles from "./ContextMenu.module.scss";
+import type { MenuData } from "./hooks/useContextMenu";
+import { useNodeAdd } from "./hooks/useNodeAdd";
+import { getOptionsAddNode } from "./utils/getOptionsAddNode";
+
+type ContextMenuProps = {
+  menuData: MenuData;
+  centerCanvas: () => void;
+  copyToClipBoard?: (target: Node<NodeData>) => Promise<void>;
```

Comment:
They are a bit different since a contextMenu will always be targeting a node, while useCopyAndPaste has a click listener that is going to fire even if the user is not hovering a node